catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/suggestion/UtmUpsCoordinateProcessor.java

```diff
@@ -76,42 +79,178 @@ public UtmUpsCoordinateProcessor(CoordinateSystemTranslator translator) {
+    final List<String> utmUpsMatches = new ArrayList<>();
     while (matcher.find()) {
       final String group = matcher.group();
       LOGGER.trace("Match found [{}]", group);
-      final String utmOrUpsText = normalizeCoordinate(group);
-      final UtmUpsCoordinate utmUps = parseUtmUpsString(utmOrUpsText);
-      if (utmUps != null) {
-        nameBuilder.append(" [ ").append(utmUps.toString()).append(" ]");
-        utmUpsCoords.add(utmUps);
-      }
+      utmUpsMatches.add(capitalizeLatBand(group));
     }
-    if (utmUpsCoords.isEmpty()) {
+
+    if (utmUpsMatches.isEmpty()) {
       LOGGER.trace("No valid UTM or UPS strings could be inferred from query [{}]", query);
-      return null;
+      return Collections.emptyList();
+    }
+
+    return utmUpsMatches.size() == 1
+        ? suggestionsForSinglePoint(utmUpsMatches.get(0))
+        : Collections.singletonList(suggestionForMultiplePoints(utmUpsMatches));
+  }
+
+  /**
+   * Generates a list of {@code LiteralSuggestion}s based on the given {@code utmUpsCoord} string.
+   * This method is only used to generate suggestions when the user enters a single UTM/UPS
+   * coordinate. The following suggestions are returned depending on the information provided in the
+   * coordinate:
+   *
+   * <ul>
+   *   <li>If the {@code utmUpsCoord} does not have a latitude band, then two suggestions are
+   *       returned - one for each hemisphere - and the hemisphere is denoted by a "N" or "S"
+   *       suffix. Example: "18 631054mE 4776851mN" -> "UTM/UPS: [ 18N 631054mE 4776851mN N ]",
+   *       "UTM/UPS: [ 18S 631054mE 4776851mN S ]"
+   *   <li>If the {@code utmpUpsCoord} has an S latitude band, then two suggestions are also
+   *       returned: one for the S latitude band coordinate and another for the Southern hemisphere
+   *       coordinate. This is because the "S" is ambiguous - it can either refer to the latitude
+   *       band or the Southern hemisphere. Example: "15S 533427mE 3796272mN" -> "UTM/UPS: [ 15S
+   *       533427mE 3796272mN ]", "UTM/UPS: [ 15S 533427mE 3796272mN S ]"
+   *   <li>If the {@code utmUpsCoord} has an N latitude band, then two suggestions are returned: one
+   *       for the N latitude band and another for the Northern hemisphere. The "N" isn't ambiguous
+   *       in this case, but the Northern hemisphere suggestion is included for consistency and to
+   *       teach users about the N/S hemisphere syntax. Example: "9N 365103mE 659568mN" -> "UTM/UPS:
+   *       [ 9N 365103mE 659568mN ]", "UTM/UPS: [ 9N 365103mE 659568mN N ]"
+   *   <li>Otherwise, one suggestion is returned. Example: "20M 48831mE09437282mN" -> "UTM/UPS: [
+   *       20M 48831mE09437282mN ]"
+   * </ul>
+   *
+   * @param utmUpsCoord the user-provided gazetteer query text. It must match the {@code
+   *     PATTERN_UTM_OR_UPS_COORDINATE} pattern.
+   * @return a list of either one or two {@LiteralSuggestion}s, depending on the latitude band of
```

Comment:
Should we change this to `@return a list of either one or two {@link LiteralSuggestion}s, depending on....`?

Reply:
done